libenv/sysinfo.c

```diff
@@ -144,7 +144,10 @@ static time_t GetBootTimeFromUptimeCommand(time_t now);
 
 /*****************************************************/
 
-void CalculateDomainName(const char *nodename, const char *dnsname, char *fqname, char *uqname, char *domain);
+void CalculateDomainName(const char *nodename, const char *dnsname,
+                         char *fqname, size_t fqname_length,
+                         char *uqname, size_t uqname_length,
+                         char *domain, size_t domain_length);
```

Comment:
The word "length" is usually used for the length of a string, i.e. up to the '\0' terminator. We usually use "size" for buffer size, which also matches to the `sizeof` operator that is usually used to pass the correct number in these function calls.

So I would rename the occurrences here to `fqname_size`, `uqname_size`, `domain_size`, and this also applies to many other places in this pull request.